numpy/core/src/multiarray/descriptor.c

```diff
@@ -1678,14 +1678,13 @@ _convert_from_str(PyObject *obj, int align)
         }
 
         /* Check for a deprecated Numeric-style typecode */
-        char *dep_tps[] = {"Bool", "Complex", "Float", "Int",
-                           "Object0", "String0", "Timedelta64",
-                           "Unicode0", "UInt", "Void0"};
+        char *dep_tps[] = {"Bytes", "Datetime64", "Str", "Uint64"};
```

Comment:
Perhaps to play it safe:
```suggestion
        /* `Uint` has deliberately weird uppercasing */
        char *dep_tps[] = {"Bytes", "Datetime64", "Str", "Uint"};
```